version: '3.9'

services:
  simtix-client-db:
    image: postgres
    env_file:
      - .env
    hostname: ${PSQL_HOST}
    environment:
      POSTGRES_DB: ${PSQL_DBNAME}
      POSTGRES_USER: ${PSQL_USERNAME}
      POSTGRES_PASSWORD: ${PSQL_PASSWORD}
    ports:
      - '8001:${PSQL_PORT}'
    volumes:
      - simtix_client_db:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${PSQL_USERNAME} -d ${PSQL_DBNAME} -h ${PSQL_HOST} -p 5432"]
      interval: 5s
      timeout: 3s
      retries: 5
    networks:
      - simtix

  simtix-client:
    build: .
    env_file:
      - .env
    depends_on:
      simtix-client-db:
        condition: service_healthy
    restart: on-failure
    networks:
      - simtix

volumes:
  simtix_client_db:

networks:
  simtix:
    external: true